mitosheet/src/components/footer/SheetTabActions.tsx

```diff
@@ -16,58 +17,112 @@ export default function SheetTabActions(props: {
     setUIState: React.Dispatch<React.SetStateAction<UIState>>;
     closeOpenEditingPopups: () => void;
     dfName: string, 
-    sheetIndex: number,
-    selectedSheetIndex: number,
     mitoAPI: MitoAPI
+    tabIDObj: {tabType: 'data', selectedIndex: number} | {tabType: 'graph', graphID: GraphID}
 }): JSX.Element {
 
     // Log opening the sheet tab actions
     useEffect(() => {
+        const selectedSheetIndexOrGraphID = props.tabIDObj.tabType === 'data' ? props.tabIDObj.selectedIndex : props.tabIDObj.graphID
         void props.mitoAPI.log(
             'clicked_sheet_tab_actions',
             {
-                sheet_index: props.sheetIndex
+                tab_type: props.tabIDObj.tabType,
+                selected_index_or_graph_id: selectedSheetIndexOrGraphID
             }
         )
     }, [])
 
     const onDelete = async (): Promise<void> => {
-        // If we are deleting the sheet index that is currently selected & it is not sheetIndex 0, update the selected sheet index
-        if (props.sheetIndex === props.selectedSheetIndex && props.selectedSheetIndex !== 0) {
-            props.setUIState(prevUIState => {
-                return {
-                    ...prevUIState,
-                    selectedSheetIndex: props.sheetIndex - 1
-                }
-            })
-        }
+        // If we are deleting a sheet tab, select the first sheet tab
+        props.setUIState(prevUIState => {
+            return {
+                ...prevUIState,
+                selectedTabType: 'data',
+                selectedSheetIndex: 0,
+                currOpenTaskpane: {type: TaskpaneType.NONE}
+            }
+        })
 
         // Close 
         props.closeOpenEditingPopups();
 
-        await props.mitoAPI.editDataframeDelete(props.sheetIndex)
+        if (props.tabIDObj.tabType === 'data') {
+            await props.mitoAPI.editDataframeDelete(props.tabIDObj.selectedIndex)
+        } else {
+            await props.mitoAPI.editGraphDelete(props.tabIDObj.graphID)
+        }
     }
 
     const onDuplicate = async (): Promise<void> => {
         // Close 
         props.closeOpenEditingPopups();
         
-        await props.mitoAPI.editDataframeDuplicate(props.sheetIndex)
+        if (props.tabIDObj.tabType === 'data') {
```

Comment:
It's kinda interesting each of these functions (and indeed, the sheet tab actions) are shared between the two different tab types. 

IMO, a better way of doing this would be to refactor the sheet tab actions to be totally different for each type - e.g. I think it would be better to have sheet tab actions for the sheet, and graph tab actions for the graph. This would lead to much simpler, much more maintainable code in the long run, as these two things continue to diverge (I think they will continue to diverge, as we're seeing with the Graph Data functionality). 

Can you rip these out into two?